src/variational/advi.jl

```diff
@@ -0,0 +1,105 @@
+"""
+    ADVI(samples_per_step = 10, max_iters = 5000)
+
+Automatic Differentiation Variational Inference (ADVI) for a given model.
+"""
+struct ADVI{AD} <: VariationalInference{AD}
+    samples_per_step # number of samples used to estimate the ELBO in each optimization step
+    max_iters        # maximum number of gradient steps used in optimization
+end
+
+ADVI(args...) = ADVI{ADBackend()}(args...)
+ADVI() = ADVI(10, 5000)
+
+alg_str(::ADVI) = "ADVI"
+
+vi(model::Model, alg::ADVI; optimizer = ADAGrad()) = begin
```

Comment:
Please see [style guide](https://turing.ml/docs/style-guide/) for function definitions. In short, please use
```
function foo(args...)

end
```
form, rather than this one.

Reply:
Yeah sorry about this; I even knew about the style guide, but then somewhere down the road I forgot about it... Sloppy on my part.

Reply:
No problem :)